File: Algorithm/Baekjoon/22_08_week_5/6443.py
def dfs(s, lst):
    if len(s) == len(S):
        print(s)
    
    for i in range(26):
        if lst[i]:
            lst[i] -= 1
            dfs(s+alpha[i], lst)
            lst[i] += 1
            

alpha = 'abcdefghijklmnopqrstuvwxyz'



N = int(input())
for _ in range(N):
    alpha_nums = [0] * 26
    S = input()
    l_S = len(S)
    for s in S:
        alpha_nums[alpha.index(s)] += 1
    dfs('', alpha_nums)

    

File: Algorithm/Baekjoon/22/22_10_week_4/6443.js
const fs = require('fs')
let input = fs.readFileSync('./test.txt').toString().trim().split('\n')
const N = input[0].split(' ').map(Number)
const alpha = 'abcdefghijklmnopqrstuvwxyz'
console.log(input)
for (let i = 1; i < N[0]+1; i++) {
    const alpha_nums = [0]*26
    const S = input[i].replace(/\r/g, '').split('')
    // for (s of S) {
    //     alpha_nums[]
    // }

}
